fe/fe-core/src/main/java/org/apache/doris/load/loadv2/SparkLauncherMonitors.java

```diff
@@ -0,0 +1,216 @@
+                        SparkLoadAppHandle.State oldState = handle.getState();
+                        SparkLoadAppHandle.State newState = oldState;
+                        // 1. state
+                        String state = regexGetState(line);
+                        if (state != null) {
+                            YarnApplicationState yarnState = YarnApplicationState.valueOf(state);
+                            newState = fromYarnState(yarnState);
+                            if (newState != oldState) {
+                                handle.setState(newState);
+                            }
+                        }
+                        // 2. appId
+                        String appId = regexGetAppId(line);
+                        if (appId != null) {
+                            if (!appId.equals(handle.getAppId())) {
+                                handle.setAppId(appId);
+                            }
+                        }
+
+                        LOG.info("spark appId that handle get is {}, state: {}", handle.getAppId(), handle.getState().toString());
+                        switch (newState) {
+                            case UNKNOWN:
+                            case CONNECTED:
+                            case SUBMITTED:
+                                // If the app stays in the UNKNOWN/CONNECTED/SUBMITTED state for more than submitTimeoutMs
+                                // stop monitoring and kill the process
+                                if (System.currentTimeMillis() - startTime > submitTimeoutMs) {
+                                    isStop = true;
+                                    handle.kill();
+                                }
+                                break;
+                            case RUNNING:
+                            case FINISHED:
+                                // There's no need to parse all logs of handle process to get all the information.
+                                // As soon as the state changes to RUNNING/KILLED/FAILED/FINISHED/LOST,
+                                // stop monitoring but keep the process alive.
+                                isStop = true;
+                                break;
+                            case KILLED:
+                            case FAILED:
+                            case LOST:
+                                // If the state changes to KILLED/FAILED/LOST,
+                                // stop monitoring and kill the process
+                                isStop = true;
+                                handle.kill();
+                                break;
+                            default:
+                                Preconditions.checkState(false, "wrong spark app state");
+                        }
+                    }
+                    // parse other values
+                    else if (line.contains(QUEUE) || line.contains(START_TIME) || line.contains(FINAL_STATUS) ||
```

Comment:
if the line contains "STATE", the while loop may be broken. So how to guarantee that this `else if` block can be ran?

Reply:
The state's changing follows the rule of 
1. `submited > running > finished / failed` 
2. `submitted > killed`
3. `submitted > running > killed`
 Normally, the laucher will print the `queue`、`start time`、`final status`、`tracking url`、`user` logs in state submitted/runnning. So in case 2, the else if block may still not be ran when the while loop be broken.
But it's not very terrible that this else if block not be ran, beacuse the necessary value we need only contains `appId` and `state`. In this case, `queue`、`start time`、`final status`、`tracking url`、`user` is just missing.